app/lib/content_security_policy.rb

```diff
@@ -13,8 +13,44 @@ def media_host
     cdn_host_value || assets_host
   end
 
+  def sso_host
+    @sso_host ||= sso_host_from_config
+  end
+
   private
 
+  def sso_host_from_config
```

Comment:
I believe the logic here is preserved from the previous setup. The spec coverage added of all the various env var cases is strong, coverage for the specific devise options is lacking both because a) a lot of it happens on initialization and is awkward to spec/stub, b) the data structures for each option are slightly different which also made it harder to stub.